File: app/src/main/java/ch/unstable/ost/database/model/QueryHistory.java
package ch.unstable.ost.database.model;



import androidx.annotation.NonNull;
import androidx.room.Embedded;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import java.util.Date;

import ch.unstable.ost.api.model.ConnectionQuery;

import static com.google.common.base.Preconditions.checkNotNull;

@Entity(tableName = QueryHistory.TABLE_NAME)
public class QueryHistory {

    public static final String TABLE_NAME = "queries";

    @PrimaryKey(autoGenerate = true)
    private final long id;

    @NonNull
    private final Date creationDate;

    @NonNull
    private final ConnectionQuery query;

    public QueryHistory(long id, Date creationDate, ConnectionQuery query) {
        this.id = id;
        this.creationDate = checkNotNull(creationDate, "creationDate");
        this.query = checkNotNull(query, "query");
    }

    @Ignore
    public QueryHistory(ConnectionQuery query) {
        this.id = 0;
        this.creationDate = new Date();
        this.query = checkNotNull(query, "query");
    }


    public long getId() {
        return id;
    }

    @NonNull
    public ConnectionQuery getQuery() {
        return query;
    }

    @NonNull
    public Date getCreationDate() {
        return creationDate;
    }
}

File: app/src/main/java/ch/unstable/ost/database/model/QueryHistory.kt
package ch.unstable.ost.database.model

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import ch.unstable.ost.api.model.ConnectionQuery
import com.google.common.base.Preconditions
import java.util.*

@Entity(tableName = QueryHistory.TABLE_NAME)
class QueryHistory {
    @PrimaryKey(autoGenerate = true)
    val id: Long
    val creationDate: Date
    val query: ConnectionQuery

    constructor(id: Long, creationDate: Date, query: ConnectionQuery) {
        this.id = id
        this.creationDate = Preconditions.checkNotNull(creationDate, "creationDate")
        this.query = Preconditions.checkNotNull(query, "query")
    }

    @Ignore
    constructor(query: ConnectionQuery) {
        id = 0
        creationDate = Date()
        this.query = Preconditions.checkNotNull(query, "query")
    }

    companion object {
        const val TABLE_NAME = "queries"
    }
}